lib/spack/docs/workflows.rst

```diff
@@ -0,0 +1,1150 @@
+dependencies.  In this case, you might use:
+
+.. code-block:: console
+
+    $ spack view symlink -d no cmake
+
+
+"""""""""""""""""""""""
+Hybrid Filesystem Views
+"""""""""""""""""""""""
+
+Although filesystem views are usually created by Spack, users are free
+to add to them by other means.  For example, imagine a filesystem
+view, created by Spack, that looks something like:
+
+.. code-block:: console
+
+  /path/to/MYVIEW/bin/programA -> /path/to/spack/.../bin/programA
+  /path/to/MYVIEW/lib/libA.so -> /path/to/spack/.../lib/libA.so
+
+Now, the user may add to this view by non-Spack means; for example, by
+running a classic install script.  For example:
+
+.. code-block:: console
+
+  tar -xf B.tar.gz
+  cd B/
+  ./configure --prefix=/path/to/MYVIEW \
+              --with-A=/path/to/MYVIEW
+  make && make install
+
+The result is a hybrid view:
+
+.. code-block:: console
+
+  /path/to/MYVIEW/bin/programA -> /path/to/spack/.../bin/programA
+  /path/to/MYVIEW/bin/programB
+  /path/to/MYVIEW/lib/libA.so -> /path/to/spack/.../lib/libA.so
+  /path/to/MYVIEW/lib/libB.so
+
+In this case, real files coexist, interleaved with the "view"
+symlinks.  At any time one can delete ``/path/to/MYVIEW`` or use
+``spack view`` to manage it surgically.  None of this will affect the
+real Spack install area.
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Discussion: Running Binaries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Modules, extension packages and filesystem views are all ways to
+assemble sets of Spack packages into a useful environment.  They are
+all semantically similar, in that conflicting installed packages
+cannot simultaneously be loaded, activated or included in a view.
+
+With all of these approaches, there is no guarantee that the
+environment created will be consistent.  It is possible, for example,
+to simultaneously load application A that uses OpenMPI and application
+B that uses MPICH.  Both applications will run just fine in this
+inconsistent environment because they rely on RPATHs, not the
+environment, to find their dependencies.
+
+In general, environments set up using modules vs. views will work
+similarly.  Both can be used to set up ephemeral or long-lived
+testing/development environments.  Operational differences between the
+two approaches can make one or the other preferable in certain
+environments:
+
+* Filesystem views do not require environment module infrastructure.
+  Although Spack can install ``environment-modules``, users might be
+  hostile to its use.  Filesystem views offer a good solution for
+  sysadmins serving users who just "want all the stuff I need in one
+  place" and don't want to hear about Spack.
+
+* Although modern build systems will find dependencies wherever they
+  might be, some applications with hand-built make files expect their
+  dependencies to be in one place.  One common problem is makefiles
+  that assume that ``netcdf`` and ``netcdf-fortran`` are installed in
```

Comment:
Sounds like a painful "solution."  Have you tried filesystem views here, do they work for this problem in practice?  Or is it possible to change the app software's build?  (Assumptions that all netcdf packages will be found together are just no longer valid, ever since NetCDF split up the packages.)